src/blockdev.rs

```diff
@@ -431,6 +431,10 @@ impl Mount {
         // the partition contents if umount failed
         let mountpoint = tempdir.into_path();
 
+        // ensure we're in a private mount namespace so the mount isn't
+        // visible to the rest of the system
+        sched::unshare(sched::CloneFlags::CLONE_NEWNS).context("unsharing mount namespace")?;
```

Comment:
Just to hedge our bets re.

> In contrast to the discussion in #208, we hard-fail if unshare fails. CAP_SYS_ADMIN is required for both mount() and unshare(CLONE_NEWNS), and we recommend running containers privileged, so there shouldn't be any expected unshare failures.

should we add an env var to allow skipping this call?

Also, WDYT about using std::sync::Once for this? AFAIK, it's harmless to call multiple times, but it's not necessary either and could be misleading.

Reply:
With `Once`, the code turns into this:

```rust
        static UNSHARE_INIT: Once = Once::new();
        static mut UNSHARE_ERR: Option<String> = None;
        // Once handles synchronization, so mutable static is safe
        unsafe {
            UNSHARE_INIT.call_once(|| {
                // anyhow::Error isn't Copy; convert to String
                UNSHARE_ERR = sched::unshare(sched::CloneFlags::CLONE_NEWNS)
                    .context("unsharing mount namespace")
                    .map_err(|e| e.to_string())
                    .err();
            });
            match UNSHARE_ERR {
                None => (),
                Some(ref e) => bail!(e),
            }
        }
```

I think I'd rather stick with `// multiple unshare calls are safe`.  WDYT?

Reply:
Yuck, the error handling goop here is unfortunate.

> I think I'd rather stick with // multiple unshare calls are safe. WDYT?

SGTM and thanks for investigating!

Reply:
Added environment variable and filed https://github.com/coreos/coreos-installer/issues/557 to remove it again.